internal/pkg/gateway/commit/notifier.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2021 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit
+
+import (
+	commonProto "github.com/hyperledger/fabric-protos-go/common"
+	ordererProto "github.com/hyperledger/fabric-protos-go/orderer"
+	peerProto "github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/hyperledger/fabric/common/ledger/blockledger"
+)
+
+// Notifier enables notification of commits to a peer ledger.
+type Notifier struct {
+	reader BlockReader
+}
+
+// NewNotifier returns a new commit notifier that obtains channels from a given channel factory.
+func NewNotifier(reader BlockReader) *Notifier {
+	return &Notifier{
+		reader: reader,
+	}
+}
+
+// Notify the supplied channel when the specified transaction is committed to a channel's ledger.
+func (notifier *Notifier) Notify(channelID string, transactionID string) (<-chan peerProto.TxValidationCode, error) {
+	// TODO: add a context parameter to enable the notifier to be cancelled
+	// TODO: pooling of ledger iterators over multiple invocations
+	blockIterator, err := notifier.reader.Iterator(channelID, seekNextCommit())
+	if err != nil {
+		return nil, err
+	}
+
+	commitChannel := make(chan peerProto.TxValidationCode, 1)
+	go readCommit(commitChannel, blockIterator, transactionID)
+
+	return commitChannel, nil
+}
+
+func readCommit(commit chan<- peerProto.TxValidationCode, blockIterator blockledger.Iterator, transactionID string) {
+	defer blockIterator.Close()
+	defer close(commit)
+
+	for {
+		block, status := blockIterator.Next()
+		if status != commonProto.Status_SUCCESS {
+			return
```

Comment:
Since you noted it below, I would likewise log a warning here, if the iterator fails, this is an unexpected error condition.